### PR TITLE
Closes #239

### LINKED ISSUE #239: Py_SetPythonHome ensures tests will not run if compiling in a virtualenv
Should be set to ``sys.base_prefix`` rather than ``sys.prefix``

### PR DESCRIPTION
**Pull request by mdavezac**

### CODE REVIEW

File: cmake/BemppFindDependencies.cmake [hunk 1]
```diff
@@ -28,7 +28,7 @@ find_package(Doxygen)
 find_package(CoherentPython REQUIRED)
 # Get Python home directory
 include(CallPython)
-call_python(PYTHON_HOME "import sys; print(sys.prefix)")
+call_python(PYTHON_HOME "import sys; print('%s:%s' % (sys.base_prefix, sys.base_exec_prefix))")
```
Comment (jenshnielsen) [inline cmake/BemppFindDependencies.cmake]: Does this work in python 2? https://docs.python.org/3/library/sys.html#sys.exec_prefix

Comment (mdavezac) [inline cmake/BemppFindDependencies.cmake]: Good point. Will look for something for older pythons.

Comment (mdavezac) [inline cmake/BemppFindDependencies.cmake]: Should work for python 2 and python >= 3.3

**Comment by mdavezac** [comment]
retest this please

**Comment by jenshnielsen** [comment]
retest this please

**Comment by jenshnielsen** [comment]
retest this please

**Comment by jenshnielsen** [comment]
@mdavezac Tests passes including python 2 and 3 http://development.rc.ucl.ac.uk/jenkins/job/bempp-pull-request/25/

**Pull request merged**